op-e2e/op_geth_test.go

```diff
@@ -70,3 +71,108 @@ func TestInvalidDepositInFCU(t *testing.T) {
 	require.Nil(t, err)
 	require.Equal(t, 0, balance.Cmp(common.Big0))
 }
+
+func TestBedrockSystemTxUsesZeroGas(t *testing.T) {
+	cfg := DefaultSystemConfig(t)
+	cfg.DeployConfig.FundDevAccounts = false
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+	opGeth, err := NewOpGeth(t, ctx, &cfg)
+	require.NoError(t, err)
+	defer opGeth.Close()
+
+	block, err := opGeth.AddL2Block(ctx)
+	require.NoError(t, err)
+	infoTx, err := opGeth.L2Client.TransactionInBlock(ctx, block.BlockHash, 0)
+	require.NoError(t, err)
+	require.True(t, infoTx.IsSystemTx())
+	receipt, err := opGeth.L2Client.TransactionReceipt(ctx, infoTx.Hash())
+	require.NoError(t, err)
+	require.Zero(t, receipt.GasUsed)
+}
+
+func TestBedrockDepositTx(t *testing.T) {
+	cfg := DefaultSystemConfig(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+	opGeth, err := NewOpGeth(t, ctx, &cfg)
+	require.NoError(t, err)
+	defer opGeth.Close()
+
+	aliceAddr := cfg.Secrets.Addresses().Alice
+
+	// Deposit TX with a higher gas limit than required
+	depositTx := types.NewTx(&types.DepositTx{
+		SourceHash:          opGeth.L1Head.Hash(),
```

Comment:
This technically isn't the correct way to compute the source hash. It doesn't impact the test but it also is a little confusing

Reply:
ah interesting.  Previously this was just left nil with a TODO.  I could add a helper function to `OpGeth` to generate this correctly but it would wind up duplicating some code from the `derive` pipeline. So given the SourceHash isn't actually used I'll go back to just omitting `SourceHash`. If it becomes important in the future `OpGeth` could be updated to automatically set it to something reasonable by default which keeps the test code minimal.